src/robot_state_publisher.cpp

```diff
@@ -147,7 +147,20 @@ RobotStatePublisher::RobotStatePublisher(const rclcpp::NodeOptions & options)
     std::bind(&RobotStatePublisher::callbackJointState, this, std::placeholders::_1),
     subscriber_options);
 
-  publishFixedTransforms();
+  double tf_static_publish_freq = this->declare_parameter("tf_static_publish_frequency", 0.0);
+  if (tf_static_publish_freq > 1000.0 || tf_static_publish_freq < 0.0) {
+    throw std::runtime_error("tf_static_publish_frequency must be between 0 and 1000");
+  }
+
+  if (tf_static_publish_freq != 0.0) {
```

Comment:
easier to comprehension would be the comparison bigger than `if (tf_static_publish_freq > 0.0) {`